### PR TITLE
Improve component docs

### PR DESCRIPTION
**Pull request by tigerros**
This removes all of these:
```
# Props

See [`FooProps`].
```
Dioxus 0.5 has made the docs a lot cleaner. Take a look at this:
image
The `props` argument is pretty obvious and the "Props" section is IMO unnecessary.

### CODE REVIEW

**Comment by tigerros** [comment]
In fact, why is the first line of every component doc "`Foo` component."? Really weird

**Comment by tigerros** [comment]
I've also removed the other unnecessary docs, specifically "`Foo` component." and "`Foo` component properties."

File: crates/components/src/popup.rs [hunk 1]
```diff
@@ -20,8 +20,6 @@ pub fn PopupBackground(children: Element) -> Element {
     })
 }
 
-/// `Popup` component.
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/popup.rs:23]:
```suggestion
/// Floating window intended for quick interactions. Also called `Dialog` in other frameworks.
///
/// # Styling
```

File: crates/components/src/scroll_views/scroll_view.rs [hunk 2]
```diff
@@ -31,11 +30,6 @@ pub struct ScrollViewProps {
     pub scroll_with_arrows: bool,
 }
 
-/// `ScrollView` component.
-///
-/// # Props
-/// See [`ScrollViewProps`](ScrollViewProps).
-///
 /// # Example
```
Comment (marc2332) [inline crates/components/src/scroll_views/scroll_view.rs:33]:
```suggestion
/// Scrollable area with bidirectional support and scrollbars.
///
/// # Example
```

File: crates/components/src/switch.rs [hunk 3]
```diff
@@ -31,9 +30,6 @@ pub enum SwitchStatus {
 
 /// Controlled `Switch` component.
```
Comment (marc2332) [inline crates/components/src/switch.rs:31]:
```suggestion
/// Display whether a state is `true` or `false`. Commonly used for enabled/disabled scenarios. Example: Light/Dark theme.
```

File: crates/components/src/table.rs [hunk 4]
```diff
@@ -203,14 +181,8 @@ pub struct TableProps {
     pub children: Element,
 }
 
-/// `Table` component.
-///
-/// # Props
-/// See [`TableProps`].
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/table.rs:184]:
```suggestion
/// Table component, composed with [`TableHead`] and [`TableBody`].
///
/// # Styling
```

File: crates/components/src/table.rs [hunk 5]
```diff
@@ -144,10 +127,6 @@ pub struct TableCellProps {
 }
 
 /// `TableCell` component.
```
Comment (marc2332) [inline crates/components/src/table.rs:129]:
```suggestion
/// Cell for a [`Table`]. You can place any other element inside.
```

File: crates/components/src/table.rs [hunk 6]
```diff
@@ -70,11 +59,6 @@ pub struct TableRowProps {
     alternate_colors: bool,
 }
 
-/// `TableRow` component.
-///
-/// # Props
-/// See [`TableRowProps`].
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/table.rs:62]:
```suggestion
/// Table row for [`Table`]. Use [`TableCell`] inside.
///
/// # Styling
```

File: crates/components/src/table.rs [hunk 7]
```diff
@@ -20,45 +20,34 @@ fn TableArrow(order_direction: OrderDirection) -> Element {
...
 }
 
-/// [`TableBody`] component properties.
 #[derive(Props, Clone, PartialEq)]
 pub struct TableBodyProps {
     /// The content of this table body.
     pub children: Element,
 }
 
 /// `TableBody` component.
```
Comment (marc2332) [inline crates/components/src/table.rs:43]:
```suggestion
/// The body of a [`Table`].
```

File: crates/components/src/table.rs [hunk 8]
```diff
@@ -20,45 +20,34 @@ fn TableArrow(order_direction: OrderDirection) -> Element {
...
 }
 
-/// [`TableHead`] component properties.
 #[derive(Props, Clone, PartialEq)]
 pub struct TableHeadProps {
     /// The content of this table head.
     pub children: Element,
 }
 
 /// `TableHead` component.
```
Comment (marc2332) [inline crates/components/src/table.rs:29]:
```suggestion
/// The head of a [`Table`]. Use [`TableRow`] inside.
```

File: crates/components/src/tile.rs [hunk 9]
```diff
@@ -13,8 +13,6 @@ pub enum TileStatus {
     Hovering,
 }
 
-/// `Tile` component.
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/tile.rs:16]:
```suggestion
/// Tile component to be used with [`Radio`] and [`Checkbox`].
/// # Styling
```

File: crates/components/src/gesture_area.rs [hunk 10]
```diff
@@ -37,11 +36,6 @@ pub struct GestureAreaProps {
 
 type EventsQueue = VecDeque<(Instant, TouchEvent)>;
 
-/// `GestureArea` component.
-///
-/// # Props
-/// See [`GestureAreaProps`].
-///
 /// # Example
```
Comment (marc2332) [inline crates/components/src/gesture_area.rs:39]:
```suggestion
/// Detect complex touch gestures such as `DoubleTap`.
///
/// # Example
```

File: crates/components/src/input.rs [hunk 11]
```diff
@@ -50,11 +49,6 @@ pub struct InputProps {
     pub mode: InputMode,
 }
 
-/// `Input` component.
-///
-/// # Props
-/// See [`InputProps`].
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/input.rs:52]:
```suggestion
/// Small box to edit text.
///
/// # Styling
```

File: crates/components/src/accordion.rs [hunk 12]
```diff
@@ -29,11 +28,6 @@ pub struct AccordionProps {
     pub summary: Element,
 }
 
-/// `Accordion` component.
-///
-/// # Props
-/// See [`AccordionProps`].
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/accordion.rs:31]:
```suggestion
/// Show other elements under a collapsable box.
///
/// # Styling
```

File: crates/components/src/button.rs [hunk 13]
```diff
@@ -26,11 +25,6 @@ pub enum ButtonStatus {
     Hovering,
 }
 
-/// `Button` component.
-///
-/// # Props
-/// See [`ButtonProps`].
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/button.rs:28]:
```suggestion
/// Clickable button.
///
/// # Styling
```

File: crates/components/src/cursor_area.rs [hunk 14]
```diff
@@ -12,11 +11,6 @@ pub struct CursorAreaProps {
     children: Element,
 }
 
-/// `CursorArea` component.
-///
-/// # Props
-/// See [`CursorAreaProps`].
-///
 /// # Example
```
Comment (marc2332) [inline crates/components/src/cursor_area.rs:14]:
```suggestion
/// Change the cursor icon when it's hovering this component.
///
/// # Example
```

File: crates/components/src/dropdown.rs [hunk 15]
```diff
@@ -145,11 +138,6 @@ pub enum DropdownStatus {
     Hovering,
 }
 
-/// `Dropdown` component.
-///
-/// # Props
-/// See [`DropdownProps`].
-///
 /// # Styling
```
Comment (marc2332) [inline crates/components/src/dropdown.rs:141]:
```suggestion
/// Select from multiple options, use alongside [`DropdownItem`].
///
/// # Styling
```

**Comment by tigerros** [comment]
Accidentally added a `use` statement which messed with the type system, should be good now

**Comment by marc2332** [comment]
We should add back the component references in the props structs, otherwise the components not using the #[component] macro won't have any "summary", see:

imagen

File: crates/components/src/accordion.rs [hunk 16]
```diff
@@ -18,7 +18,6 @@ pub enum AccordionStatus {
     Hovering,
 }
 
-/// [`Accordion`] component properties.
```
Comment (marc2332) [inline crates/components/src/accordion.rs:21]: Like this one

**Comment by marc2332** [comment]
Nice! I think I will do another PR doing some final touches soon, thanks!

**Pull request merged**